src/Commands/MeltorCommand.php

```diff
@@ -0,0 +1,309 @@
+            }
+
+            // Foreign Keys.
+            foreach ($foreignKeys->get($tableName)?->groupBy('CONSTRAINT_NAME') ?? [] as $foreignKey) {
+                $foreignKeyMigration = $this->meltor->generateForeignKeyMigration($foreignKey, $this->ignoreProblems, $this);
+                if ($this->separateForeignKeys) {
+                    $columnConstraintMigrations[] = $foreignKeyMigration;
+                } else {
+                    $columnMigrations[] = $foreignKeyMigration;
+                }
+            }
+
+            $tableMigrations[$tableName]['laravel'] = $columnMigrations;
+            $tableMigrations[$tableName]['raw']     = $columnMigrationsRaw;
+
+            if ($this->separateForeignKeys) {
+                $constraintMigrations[$tableName] = $columnConstraintMigrations;
+            }
+        }
+
+        $migrationCode      = $this->meltor->generateMigration($tableMigrations, $this->meltor->getMigrationTemplate('migrationComment'), $constraintMigrations, $rawExceptions);
+        $migrationCode      = $this->meltor->beautify($migrationCode);
+        $migrationFilePath  = $this->meltor->writeMigration($migrationCode, $this->meltor->config('migration.name'), $this);
+        $showDisclaimerText = true;
+
+        if ($this->option('testrun')) {
+            $showDisclaimerText = $this->testrun();
+        }
+
+        if ($showDisclaimerText) {
+            $this->newLine();
+            $this->line(
+                sprintf('Test and inspect "%s" to make sure the migration fits your requirements', $migrationFilePath)
+            );
+            $this->line('You can then delete all prior migrations and commit');
+            $this->warn('It may be necessary to keep migrations that alter tables created by frameworks or packages!');
+        }
+
+        if ($this->warnAboutFloat) {
+            $this->warn('FLOAT columns may be turned to DOUBLE by Laravel!');
+        }
+
+        $this->newLine();
+
+        return 0;
+    }
+
+    /**
+     * Compares the database structure before and after the new migration.
+     *
+     * Returns true success, but also if there are still changes to review.
+     * Returns false on complete failures.
+     *
+     * @return bool
+     */
+    protected function testrun(): bool
+    {
+        $success                 = false;
+        $beforeStructureFileName = $this->meltor->combinePath(
+            $this->meltor->config('testrun.folder'),
+            $this->meltor->config('testrun.beforeStructureFileName')
+        );
+        $afterStructureFileName  = $this->meltor->combinePath(
+            $this->meltor->config('testrun.folder'),
+            $this->meltor->config('testrun.afterStructureFileName')
+        );
+
+        // Reading existing database structure.
+        $this->newLine();
+        $this->line('Performing structure comparison test run');
+        $this->meltor->writeStructureDump($beforeStructureFileName, $this);
```

Comment:
I don't know, I'm not a fan of passing the Command instance to the Meltor object each time. This could also be part of the initialization, and then probably only `$this->getOutput()` and not the whole Command. Preferably, register a logging callback on construct that you call if it is defined. Alternatively, you could use Events or Notifications.